src/metrics/mongoose_metrics_api.erl

```diff
@@ -16,30 +16,59 @@
     {ok, #{binary() => binary() | [metric_dict_result()]}}
     | {error, binary()}.
 
--spec get_metrics(Name :: name()) -> {ok, [metric_result()]}.
+-spec get_metrics(Name :: name()) -> {ok, [metric_result()]} | {no_name_found, binary()}.
 get_metrics(Name) ->
-    Values = exometer:get_values(Name),
-    {ok, lists:map(fun make_metric_result/1, Values)}.
+    case exometer:get_values(Name) of
+        [] ->
+            {no_name_found, <<"No value found for queried name">>};
+        Values ->
+            {ok, lists:map(fun make_metric_result/1, Values)}
+    end.
 
 -spec get_metrics_as_dicts(Name :: name(), Keys :: [key()]) ->
-    {ok, [metric_dict_result()]}.
+    {ok, [metric_dict_result()]} | {no_name_found | no_key_found, binary()}.
 get_metrics_as_dicts(Name, Keys) ->
-    Values = exometer:get_values(Name),
-    {ok, [make_metric_dict_result(V, Keys) || V <- Values]}.
+    case exometer:get_values(Name) of
+        [] ->
+            {no_name_found, <<"No value found for queried names">>};
+        Values ->
+            Results = [make_metric_dict_result({N, Dict}, Keys) ||
+                       {N, Dict} <- Values, filter_keys(Dict, Keys) =/= []],
```

Comment:
it changes logic.
Before, we were still returning metric even if keys are not matching, which is pretty logical, could be used to list metric names and consistent between other exometer APIs.

Now you just silent remove a metric from the result set, if it has no keys.